tools/common/commoncli/cli.go

```diff
@@ -0,0 +1,218 @@
+// The MIT License (MIT)
+
+// Copyright (c) 2017-2020 Uber Technologies Inc.
+
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files (the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions:
+//
+// The above copyright notice and this permission notice shall be included in all
+// copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+// SOFTWARE.
+
+package commoncli
+
+import (
+	"errors"
+	"fmt"
+	"io"
+	"os"
+	"strings"
+
+	"github.com/fatih/color"
+	"github.com/urfave/cli/v2"
+)
+
+var (
+	colorRed     = color.New(color.FgRed).SprintFunc()
+	colorMagenta = color.New(color.FgMagenta).SprintFunc()
+)
+
+func New(name, usage, version string) *cli.App {
+	if version == "" {
+		version = "0.0.1" // common "unversioned" value across many, tho likely worth dropping
+	}
+
+	app := cli.NewApp()
+	app.Name = name
+	app.Usage = usage
+	app.Version = version
+	app.ExitErrHandler = exitHandler
```

Comment:
Test is failing due to assumption that app will not exit before returning error. 

https://github.com/uber/cadence/blob/d19937f56d7de665a6f2b523da88fe806381b0b4/tools/common/schema/test/setuptest.go#L72

Reply:
aaah, I see:
```go
tb.Error(app.Run([]string{"./tool", ...}), "make sure command doesn't succeed without version or disable-version")
```
yea, makes sense.  this is handled better by the new code too.